gno.land/pkg/gnoclient/mock.go

```diff
@@ -0,0 +1,285 @@
+package gnoclient
+
+import (
+	"github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	ctypes "github.com/gnolang/gno/tm2/pkg/bft/rpc/core/types"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/crypto/hd"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+// Signer mock
+type (
+	mockSign     func(cfg SignCfg) (*std.Tx, error)
+	mockInfo     func() keys.Info
+	mockValidate func() error
+)
+
+type mockSigner struct {
+	sign     mockSign
+	info     mockInfo
+	validate mockValidate
+}
+
+func (m *mockSigner) Sign(cfg SignCfg) (*std.Tx, error) {
+	if m.sign != nil {
+		return m.sign(cfg)
+	}
+	panic("no implementation passed in")
```

Comment:
Please, remove the panics in these mock implementations 🙏 

The rule of thumb is to always return a nil or empty object, not panic if the mock function is not defined

Reply:
[44a0657](https://github.com/gnolang/gno/pull/1565/commits/44a0657e4cfa7af44b458cc9867ab8196a17f1fa)